ball_detect/main.c

```diff
@@ -17,14 +17,22 @@
 #define DIP_SWITCH_5 4
 #define DIP_SWITCH_6 5
 #define F_OSC F_CPU
+#define NOT_SEEING_VALUE 32000
+#define QUANTITY_OF_SENSORS 4
 #define ULTRASONIC_SENSOR 4
+#define TIME_WITHOUT_PULSE 100000
+#define TIMER_OVERFLOW 2000000
 #define waitForTX() while (!(UCSR0A & 1<<UDRE0))
 
-volatile int32_t ends_of_pulses[4];
-volatile int32_t lenghts_of_pulses[4];
-volatile int32_t starts_of_pulses[4]; 
-volatile int32_t vision_result[3];
-volatile uint8_t pinstate, ct, changed_bits, portb_history = 0xFF, my_address;
+volatile int32_t not_seeing[QUANTITY_OF_SENSORS];
+volatile int32_t ends_of_pulses[QUANTITY_OF_SENSORS];
+volatile int32_t lenghts_of_pulses[QUANTITY_OF_SENSORS];
+volatile int32_t starts_of_pulses[QUANTITY_OF_SENSORS]; 
+volatile int32_t vision_result[QUANTITY_OF_SENSORS];
+volatile int32_t sensor_values[QUANTITY_OF_SENSORS];
+volatile int32_t counter[QUANTITY_OF_SENSORS];
+volatile uint32_t now;
+volatile uint8_t pinstate, ct, changed_bits, portb_history = 0xFF, my_address, address_of_message = 0;
```

Comment:
I'm not sure but I think this message might be more than 80 characters long. Could you please check?